main/cicd/cicd-pipeline/config/infra/cloudformation.yml

```diff
@@ -199,6 +199,115 @@ Resources:
                 Effect: Allow
                 Resource: !Ref PipelineNotificationsTopic
 
+  PolicyAppDeployer:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Allows CodePipeline to create resources required for deploying SWB
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Action:
+              - iam:GetRole
+              - iam:GetRolePolicy
+              - iam:PassRole
+              - iam:CreateRole
+              - iam:PutRolePolicy
+              - iam:AttachRolePolicy
+              - iam:DeleteRole
+              - iam:DeleteRolePolicy
+              - iam:DetachRolePolicy
+              - iam:*TagRole*
+            Effect: Allow
+            Resource: !Sub 'arn:aws:iam::${AWS::AccountId}:role/${self:custom.settings.namespace}-*'
+          - Action:
+              - iam:CreatePolicy
+              - iam:GetPolicy
+              - iam:*PolicyVersion*
+            Effect: Allow
+            Resource: !Sub 'arn:aws:iam::${AWS::AccountId}:policy/${self:custom.settings.namespace}-*'
+          - Action:
+              - dynamodb:*
+            Effect: Allow
+            Resource: !Sub 'arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${self:custom.settings.namespace}-*'
+          - Effect: Allow
+            Action:
+              - logs:CreateLogStream
+              - logs:DescribeLogStreams
+              - logs:PutLogEvents
+              - logs:CreateLogGroup
+              - logs:DeleteLogGroup
+              - logs:DeleteLogStream
+            Resource:
+              - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/*'
+              - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/*:log-stream:*'
+          - Effect: Allow
+            Action:
+              - s3:*
+              - lambda:*
+              - apigateway:*
+              - cloudfront:*
+              - servicecatalog:*
+            Resource:
+              - '*'
+          - Effect: Allow
+            Action:
+              - ssm:GetParameter
+            Resource:
+              - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${self:custom.settings.envName}/*'
+          - Effect: Allow
+            Action:
+              - iam:ListRoles
+              - organizations:DescribeOrganization
+              - account:ListRegions
+              - cloudformation:ValidateTemplate
+            Resource:
+              - '*' # For the actions listed above IAM does not support resource-level permissions and requires all resources to be chosen
+          - Effect: Allow
+            Action:
+              - iam:CreateServiceLinkedRole
+              - iam:DeleteServiceLinkedRole
```

Comment:
You might also need `- iam:GetServiceLinkedRoleDeletionStatus` for managing service linked roles.